opslevel/resource_opslevel_check_repository_grep.go

```diff
@@ -1,108 +1,242 @@
-// 	input.DirectorySearch = opslevel.RefOf(d.Get("directory_search").(bool))
-
-// 	if d.HasChange("filepaths") {
-// 		input.FilePaths = opslevel.RefOf(getStringArray(d, "filepaths"))
-// 	}
-// 	if d.HasChange("file_contents_predicate") {
-// 		input.FileContentsPredicate = expandPredicateUpdate(d, "file_contents_predicate")
-// 	}
-
-// 	_, err := client.UpdateCheckRepositoryGrep(*input)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	d.Set("last_updated", timeLastUpdated())
-// 	return resourceCheckRepositoryGrepRead(d, client)
-// }
+// TODO: register resource to provider
+// TODO: write tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+	"github.com/relvacode/iso8601"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &CheckRepositoryGrepResource{}
+	_ resource.ResourceWithImportState = &CheckRepositoryGrepResource{}
+)
+
+func NewCheckRepositoryGrepResource() resource.Resource {
+	return &CheckRepositoryGrepResource{}
+}
+
+// CheckRepositoryGrepResource defines the resource implementation.
+type CheckRepositoryGrepResource struct {
+	CommonResourceClient
+}
+
+type CheckRepositoryGrepResourceModel struct {
+	Category    types.String `tfsdk:"category"`
+	Description types.String `tfsdk:"description"`
+	Enabled     types.Bool   `tfsdk:"enabled"`
+	EnableOn    types.String `tfsdk:"enable_on"`
+	Filter      types.String `tfsdk:"filter"`
+	Id          types.String `tfsdk:"id"`
+	Level       types.String `tfsdk:"level"`
+	Name        types.String `tfsdk:"name"`
+	Notes       types.String `tfsdk:"notes"`
+	Owner       types.String `tfsdk:"owner"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+
+	DirectorySearch       types.Bool      `tfsdk:"directory_search"`
+	Filepaths             types.List      `tfsdk:"filepaths"`
+	FileContentsPredicate *PredicateModel `tfsdk:"file_contents_predicate"`
+}
+
+func NewCheckRepositoryGrepResourceModel(ctx context.Context, check opslevel.Check) (CheckRepositoryGrepResourceModel, diag.Diagnostics) {
+	var model CheckRepositoryGrepResourceModel
+
+	model.Category = types.StringValue(string(check.Category.Id))
+	model.Enabled = types.BoolValue(check.Enabled)
+	model.EnableOn = types.StringValue(check.EnableOn.Time.Format(time.RFC3339))
+	model.Filter = types.StringValue(string(check.Filter.Id))
+	model.Id = types.StringValue(string(check.Id))
+	model.Level = types.StringValue(string(check.Level.Id))
+	model.Name = types.StringValue(check.Name)
+	model.Notes = types.StringValue(check.Notes)
+	model.Owner = types.StringValue(string(check.Owner.Team.Id))
+	model.LastUpdated = timeLastUpdated()
```

Comment:
I don't think this should be called here because every read would update this time.

Reply:
you are correct - 👍